specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2022-11-01-preview/DynamicSummaries.json

```diff
@@ -0,0 +1,398 @@
+        }
+      }
+    }
+  },
+  "definitions": {
+    "SummaryList": {
+      "description": "List all the summaries.",
+      "type": "object",
+      "properties": {
+        "nextLink": {
+          "description": "URL to fetch the next set of summaries.",
+          "readOnly": true,
+          "type": "string"
+        },
+        "value": {
+          "description": "Array of summaries.",
+          "items": {
+            "$ref": "#/definitions/Summary"
+          },
+          "type": "array"
+        }
+      },
+      "required": [
+        "value"
+      ]
+    },
+    "Summary": {
+      "allOf": [
+        {
+          "$ref": "../../../common/2.0/types.json#/definitions/ResourceWithEtag"
+        }
+      ],
+      "description": "Represents a Summary in Azure Security Insights.",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/SummaryProperties",
+          "description": "Summary properties",
+          "x-ms-client-flatten": true
+        }
+      },
+      "type": "object"
+    },
+    "SummaryProperties": {
+      "description": "Describes summary properties",
+      "properties": {
+        "summaryId": {
+          "description": "The ID (a GUID) of the summary",
+          "type": "string"
+        },
+        "summaryName": {
+          "description": "The name of the summary",
+          "type": "string"
+        },
+        "sourceInfo": {
+          "description": "The source info for the summary",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "searchKey": {
+          "description": "The search key is used to optimize query performance when using summaries for joins with other data. For example, enable a column with IP addresses to be the designated SearchKey field, then use this field as the key field when joining to other event data by IP address.",
+          "type": "string"
+        },
+        "tactics": {
+          "items": {
+            "$ref": "./common/AlertTypes.json#/definitions/AttackTactic"
+          },
+          "description": "A list of relevant MITRE attacks",
+          "type": "array"
+        },
+        "techniques": {
+          "description": "The list of techniques for summary",
+          "items": {
+            "description": "Related Technique",
+            "type": "string"
+          },
+          "type": "array"
+        },
+        "relationName": {
+          "description": "The relation name for the summary",
+          "type": "string"
+        },
+        "relationId": {
+          "description": "The relation ID (GUID) for the summary",
+          "type": "string"
+        },
+        "summaryDescription": {
+          "description": "The description for the summary",
+          "type": "string"
+        },
+        "rawContent": {
+          "description": "The raw content that represents to summary items to create.",
+          "type": "string"
+        },
```

Comment:
As you have already noted this is subject to the ARM limitation on the size. You should consider alternate methods of passing in such large data like a blob with SAS url etc

Reply:
we're working on adding uploading items via data plane. but this is a plan for future (maybe next semester)